remhos_lo.cpp

```diff
@@ -238,6 +239,71 @@ void ResidualDistribution::CalcLOSolution(const Vector &u, Vector &du) const
    }
 }
 
+void MassBasedAvg::CalcLOSolution(const Vector &u, Vector &du) const
+{
+   // Compute the new HO solution.
+   Vector du_HO(u.Size());
+   ParGridFunction u_HO_new(&pfes);
+   ho_solver.CalcHOSolution(u, du_HO);
+   add(1.0, u, dt, du_HO, u_HO_new);
+
+   // Mesh positions for the new HO solution.
+   ParMesh *pmesh = pfes.GetParMesh();
+   GridFunction x_new(pmesh->GetNodes()->FESpace());
+   // Copy the current nodes into x.
+   pmesh->GetNodes(x_new);
+   if (mesh_v)
+   {
+      // Remap mode - get the positions of the mesh at time [t + dt].
+      x_new.Add(dt, *mesh_v);
```

Comment:
Would you have to move the mesh back after computing the masses and volumes at the new position?

Reply:
Here `x_new` is a copy of the nodes, the mesh is not moved.